### PR TITLE
add solution for jv-fruit-shop-tests

### PR DESCRIPTION
**Pull request by SerhiiPankov**

### CODE REVIEW

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FruitDaoImplTest {
+    private final String firstName = "apple";
+    private final int firstAmount = 5;
+    private final String secondName = "banana";
+    private final int secondAmount = 10;
+    private final String thirdName = "pineapple";
+    private final int thirdAmount = 15;
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:15]: make all these variables local

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    private final FruitDao fruitDao = new FruitDaoImpl();
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:16]: Initialize this field inside of method with annotation @BeforeClass

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    @Test
+    public void addFruit_Ok() {
+        Storage.fruits.clear();
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:20]: you use this line in all test methods, move it to the method with annotation @AfterEach

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        Fruit fruitThird = new Fruit();
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:21]:
```suggestion
        Fruit fruit = new Fruit();
```

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        fruitFirst.setName(firstName);
+        fruitFirst.setAmount(firstAmount);
+        fruitDao.addFruit(fruitFirst);
+        Fruit fruitSecond = new Fruit();
+        fruitSecond.setName(secondName);
+        fruitSecond.setAmount(secondAmount);
+        fruitDao.addFruit(fruitSecond);
+        List<String> expectedList = List.of(
+                "apple,5", "banana,10");
+        List<String> getAllList = fruitDao.getAll();
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:62]:
```suggestion
        List<String> actualList = fruitDao.getAll();
```

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        Assert.assertTrue(Storage.fruits.contains(fruitThird));
+    }
+
+    @Test
+    public void getByName_Ok() {
+        Storage.fruits.clear();
+        Fruit fruitFirst = new Fruit();
+        fruitFirst.setName(firstName);
+        fruitFirst.setAmount(firstAmount);
+        fruitDao.addFruit(fruitFirst);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:34]: you use the same block of code in several places but with different values, let's create a private method, that will be creating a fruit, passing needed values

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FileReaderImplTest {
+    private static final String fileName = "temp.csv";
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileReaderImplTest.java:13]: how we should write a constant name?

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    private static final Path path = Paths.get(fileName);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileReaderImplTest.java:14]: make it local, you can't do this right here, it's a bad practice to call such logic during initialization of field

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    private static final FileReader fileReader = new FileReaderImpl();
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileReaderImplTest.java:15]: Initialize this field inside of method with annotation @BeforeClass

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Test
+    public void readFromFile_Ok() throws IOException {
+        Files.deleteIfExists(path);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileReaderImplTest.java:19]: you use this line in all test methods, move it to the method with annotation @AfterEach

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FileWriterImplTest {
+    private static final String fileName = "temp.csv";
+    private static final String wrongName = "src/temp/temp.csv";
+    private static final Path path = Paths.get(fileName);
+    private static final FileWriter fileWriter = new FileWriterImpl();
+    private static final List<String> lines = List.of(
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileWriterImplTest.java:17]: make it local

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FileWriterImplTest {
+    private static final String fileName = "temp.csv";
+    private static final String wrongName = "src/temp/temp.csv";
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileWriterImplTest.java:14]: make them local

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    private static final Path path = Paths.get(fileName);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileWriterImplTest.java:15]: make it local, you can't do this right here, it's a bad practice to call such logic during initialization of field

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    private static final FileWriter fileWriter = new FileWriterImpl();
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileWriterImplTest.java:16]: Initialize this field inside of method with annotation @BeforeClass

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+    @Test
+    public void writeToFile_OK() throws IOException {
+        File file = new File(fileName);
+        fileWriter.writeToFile(fileName, lines);
+        Assert.assertTrue(Files.exists(path));
+        List<String> linesFromFile = Files.readAllLines(file.toPath());
+        for (int i = 0; i < linesFromFile.size(); i++) {
+            Assert.assertEquals(lines.get(i), linesFromFile.get(i));
+        }
+        Files.deleteIfExists(path);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileWriterImplTest.java:38]: do you need this line?

File: src/test/java/core/basesyntax/operation/OperationTest.java [hunk 16]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class OperationTest {
+    @Test
+    public void getOperation_Ok() {
+        String operationCharacter = "b";
+        Assert.assertSame(Operation.getOperation(operationCharacter), Operation.BALANCE);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/operation/OperationTest.java:10]: let's check all operations

File: src/main/java/core/basesyntax/service/CreateFruit.java [hunk 17]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.Fruit;
+
+public interface CreateFruit {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/service/CreateFruit.java:5]:
```suggestion
public interface FruitCreator {
```

File: src/test/java/core/basesyntax/service/CreateFruitImplTest.java [hunk 18]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.Fruit;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class CreateFruitImplTest {
+    @Test
+    public void createFruit() {
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/service/CreateFruitImplTest.java:9]:
```suggestion
    public void createFruit_Ok() {
```

File: src/test/java/core/basesyntax/service/CreateFruitImplTest.java [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+        String expectedName = "apple";
+        int expectedAmount = 5;
+        String testedName = "apple";
+        String testedAmount = "5";
+        Fruit expectedFruit = new Fruit();
+        expectedFruit.setName(expectedName);
+        expectedFruit.setAmount(expectedAmount);
+        Fruit fruit = new CreateFruitImpl().createFruit(testedName, testedAmount);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/service/CreateFruitImplTest.java:17]: create field FruitCreator and initialize it inside of method with @BeforeClass

File: src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+import core.basesyntax.operation.Operation;
+import core.basesyntax.service.activity.BalanceHandler;
+import core.basesyntax.service.activity.OperationHandler;
+import core.basesyntax.service.activity.SupplyHandler;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class OperationStrategyImplTest {
+    @Test
+    public void getOperation_Ok() {
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java:14]: create NotOk test

File: src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+        MapStrategy mapStrategy = new MapStrategy();
+        FruitDao fruitDao = new FruitDaoImpl();
+        OperationStrategy operationStrategy =
+                new OperationStrategyImpl(mapStrategy.mapStrategy(fruitDao));
+        OperationHandler operationHandler = operationStrategy.get(Operation.SUPPLY);
+        OperationHandler supplyHandler = new SupplyHandler(fruitDao);
+        Assert.assertEquals(supplyHandler.getClass(), operationHandler.getClass());
+        operationHandler = operationStrategy.get(Operation.BALANCE);
+        OperationHandler balanceHandler = new BalanceHandler(fruitDao);
+        Assert.assertEquals(balanceHandler.getClass(), operationHandler.getClass());
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java:24]: check all operations

File: src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+import core.basesyntax.service.activity.OperationHandler;
+import core.basesyntax.service.activity.SupplyHandler;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class OperationStrategyImplTest {
+    @Test
+    public void getOperation_Ok() {
+        MapStrategy mapStrategy = new MapStrategy();
+        FruitDao fruitDao = new FruitDaoImpl();
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java:16]: make global field of them and initialize inside of a method with @BeforeClass

File: src/test/java/core/basesyntax/service/FruitTransactionImplTest.java [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+import core.basesyntax.model.Fruit;
+import core.basesyntax.strategy.MapStrategy;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FruitTransactionImplTest {
+    private static final MapStrategy mapStrategy = new MapStrategy();
+    private static final FruitDao fruitDao = new FruitDaoImpl();
+    private FruitTransaction fruitTransaction = new FruitTransactionImpl();
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/service/FruitTransactionImplTest.java:15]: initialize them inside of a method with @BeforeClass

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java (hunk 1 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:15]: :)

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java (hunk 2 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:16]: :)

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java (hunk 3 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:20]: :)

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java (hunk 4 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:21]: :)

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java (hunk 5 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:62]: :)

File: src/test/java/core/basesyntax/dao/FruitDaoImplTest.java (hunk 6 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/dao/FruitDaoImplTest.java:34]: :)

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java (hunk 7 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/files/FileReaderImplTest.java:13]: :)

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java (hunk 8 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/files/FileReaderImplTest.java:14]: :)

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java (hunk 9 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/files/FileReaderImplTest.java:15]: :)

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java (hunk 10 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/files/FileReaderImplTest.java:19]: :)

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java (hunk 11 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/files/FileWriterImplTest.java:17]: :)

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java (hunk 13 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/files/FileWriterImplTest.java:15]: :)

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java (hunk 14 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/files/FileWriterImplTest.java:16]: :)

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java (hunk 15 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/files/FileWriterImplTest.java:38]: :)

File: src/test/java/core/basesyntax/operation/OperationTest.java (hunk 16 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/operation/OperationTest.java:10]: :)

File: src/main/java/core/basesyntax/service/CreateFruit.java (hunk 17 above)
Comment (SerhiiPankov) [reply src/main/java/core/basesyntax/service/CreateFruit.java:5]: :)

File: src/test/java/core/basesyntax/service/CreateFruitImplTest.java (hunk 18 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/service/CreateFruitImplTest.java:9]: :)

File: src/test/java/core/basesyntax/service/CreateFruitImplTest.java (hunk 19 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/service/CreateFruitImplTest.java:17]: :)

File: src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java (hunk 21 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java:24]: :)

File: src/test/java/core/basesyntax/service/FruitTransactionImplTest.java (hunk 23 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/service/FruitTransactionImplTest.java:15]: :)

File: src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java (hunk 22 above)
Comment (SerhiiPankov) [reply src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java:16]: :)

File: src/test/java/core/basesyntax/files/FileReaderImplTest.java [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+public class FileReaderImplTest {
+    private static final String FILE_NAME = "temp.csv";
+    private static Path path;
+    private static FileReader fileReader;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        path = Paths.get(FILE_NAME);
+        fileReader = new FileReaderImpl();
+        Files.deleteIfExists(path);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileReaderImplTest.java:23]: do you need to remove file here?

File: src/test/java/core/basesyntax/files/FileWriterImplTest.java [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.Test;
+
+public class FileWriterImplTest {
+    private static final List<String> LINES = List.of(
+            "type,name,quantity",
+            "b,banana,100",
+            "b,apple,10",
+            "s,banana,100",
+            "p,banana,1");
+    private static Path path;
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/files/FileWriterImplTest.java:20]: make this variable local

File: src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java [hunk 26]
```diff
@@ -0,0 +1,75 @@
...
+import core.basesyntax.service.activity.SupplyHandler;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class OperationStrategyImplTest {
+    private static MapStrategy mapStrategy;
+    private static FruitDao fruitDao;
+    private static OperationStrategy operationStrategy;
+    private static OperationHandler operationHandler;
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java:19]: don't need to make this variable global, make it local

File: src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java [hunk 27]
```diff
@@ -0,0 +1,75 @@
...
+        operationHandler = operationStrategy.get(Operation.PURCHASE);
+        OperationHandler handler = new ReturnHandler(fruitDao);
+        Assert.assertNotEquals("Test failed! Expected operation should not be: " + handler,
+                handler.getClass(), operationHandler.getClass());
+        handler = new BalanceHandler(fruitDao);
+        Assert.assertNotEquals("Test failed! Expected operation should not be: " + handler,
+                handler.getClass(), operationHandler.getClass());
+        handler = new SupplyHandler(fruitDao);
+        Assert.assertNotEquals("Test failed! Expected operation should not be: " + handler,
+                handler.getClass(), operationHandler.getClass());
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/strategy/OperationStrategyImplTest.java:73]: I meant to add test where you pass in get null

**Review by Nick97-git — CHANGES_REQUESTED** [review]

**Review by Nick97-git — APPROVED** [review]
